### PR TITLE
docs: api reference docs

### PR DESCRIPTION
**Pull request by jyecusch**

### CODE REVIEW

File: docs/reference/api/api-httpmethods.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+The following links all redirect here:
+
+- `api.get()` - HTTP `GET`
+- `api.post()` - HTTP `POST`
+- `api.put()` - HTTP `PUT`
+- `api.delete()` - HTTP `DELETE`
+- `api.patch()` - HTTP `PATCH`
```
Comment (tjholm) [inline docs/reference/api/api-httpmethods.md:7]: Docs looks great, just this note is a bit confusing

**Review by davemooreuws — COMMENTED** [review]
@jyecusch do you plan on adding these to the docs config or we do it all in a separate story?

**Comment by jyecusch** [comment]
> @jyecusch do you plan on adding these to the docs config or we do it all in a separate story?

I'll add it - probably need a features in the website too so multiple pages can come from a single template, but have separate links in the docs site. Since get/post/put/delete etc. are all so similar there isn't a lot of value in creating separate pages (so I didn't)

File: docs/reference/api/api-httpmethods.md (hunk 1 above)
Comment (jyecusch) [reply docs/reference/api/api-httpmethods.md]: I'll create another change later to fix this using a template page or something similar

**Review by davemooreuws — APPROVED** [review]

File: doc.config.json [hunk 2]
```diff
@@ -131,9 +131,66 @@
             ]
           },
           {
-            "title": "Nitric API",
+            "title": "JavaScript",
```
Comment (tjholm) [inline doc.config.json:134]: I think Node.js may work better here, as the SDK isn't web compatible at the moment.

**Review by tjholm — APPROVED** [review]

**Pull request merged**